services/app-api/src/domain-models/contractAndRates/parseDomainData.ts

```diff
@@ -0,0 +1,86 @@
+import {
+    ContractTableWithRelations,
+    DraftContractRevisionTableWithRelations,
+    DraftContractTableWithRelations,
+} from '../../postgres/prismaTypes'
+import {
+    Contract,
+    ContractRevision,
+    contractRevisionZodSchema,
+    draftContractZodSchema,
+    contractZodSchema,
+} from './contractAndRatesZodSchema'
+import {
+    draftContractToDomainModel,
+    draftContractRevToDomainModel,
+    contractWithHistoryToDomainModel,
+} from '../../postgres/contractAndRates/prismaToDomainModel'
+
+function parseDraftContractRevision(
+    revision: DraftContractRevisionTableWithRelations
+): ContractRevision | Error {
+    const draftContractRevision = draftContractRevToDomainModel(revision)
+    const parseDraft = contractRevisionZodSchema.safeParse(
+        draftContractRevision
+    )
+
+    if (!parseDraft.success) {
+        console.warn(
+            `ERROR: attempting to parse prisma draft contract revision failed: ${parseDraft.error}`
```

Comment:
issue: Can you quickly throw in some OTEL here,  not just console.warn. Same with other errors in this file. I'm assuming we wouldn't call this function in prod unless the flag is on so not worried about flooding logs. Also check that the error also returns the top level  contract id or the rate id clearly.

Reply:
My thought was to log OTEL errors at the resolver level. 

I think if we wanted to log here, we could, but we would have to pass down the `span` from `context` into the `postgresStore`? If that is the case I would want to do that in a separate ticket because it would be some refactors in other parts of the code.